woocommerce/woocommerce
# Add E2E Tests and Refactor Product Bulk Edit

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #32307 .

This adds automated E2E tests for bulk editing taking them out of Product Edits to a new file. This refactors existent Product Edit and Product Bulk Edit tests in isolated test cases instead of using the `test.step` approach.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Please review the refactored code, run the corresponding tests and make sure they pass.

For new Product Bulk Edit tests:

`pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/product-bulk-edit.spec.js`

For former and refactored Product Edit tests:

`pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/product-edit.spec.js`



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
